src/main/scala/is/hail/variant/Call.scala

```diff
@@ -0,0 +1,147 @@
+package is.hail.variant
+
+import is.hail.check.Gen
+import is.hail.expr.{TInt, TStruct, Type}
+import is.hail.utils._
+import is.hail.variant.GenotypeType.GenotypeType
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.types._
+import org.json4s._
+
+trait Call {
```

Comment:
It was natural to model this on the Genotype interface, but I think you can make it much lighter weight.  Make the runtime value of a Call a (boxed) Int.  Then these functions should move to a Call object and take a java.lang.Integer.